server/market/orderrouter.go

```diff
@@ -534,39 +534,45 @@ func (r *OrderRouter) checkPrefixTrade(user account.AccountID, tunnel MarketTunn
 				len(coin.PubKeys), sigCount, i,
 			))
 		}
+		// Validate the coin ID and get an asset-specific string representation
+		// of the coin for error messages.
+		coinStr, err := assets.funding.Backend.CoinIDString(coin.ID)
+		if err != nil {
+			return errSet(msgjson.FundingError,
+				fmt.Sprintf("invalid coin %v", coin.ID))
+		}
 		// Check that the outpoint isn't locked.
 		locked := tunnel.CoinLocked(coinAssetID, order.CoinID(coin.ID))
 		if locked {
 			return errSet(msgjson.FundingError,
-				fmt.Sprintf("coin %x is locked", coin.ID))
+				fmt.Sprintf("coin %s is locked", coinStr))
 		}
 		// Get the coin from the backend and validate it.
 		dexCoin, err := assets.funding.Backend.FundingCoin(coin.ID, coin.Redeem)
```

Comment:
@buck54321 Ah, so you're suggesting `dexCoin.TxID()` as the equivalent here?  That would work I think although the trouble I ran into was with `CoinLocked`.  A lingering server-side order that should have been failed and coins unlocked still had coins locked and the message from CoinLocked back on the client was not legible. I suppose `CoinLocked` could simply be run after `FundingCoin`.

Reply:
Nope. I'm suggesting adding a `String` method to `asset.Coin` that prints the coin ID in a readable, asset-dependent way, probably txid:vout for our current assets.

Reply:
Ok, this works out with `asset.Coin` and thus `asset.FundingCoin` as a `Stringer`.  Practically that meant implementing `String` for both the `UTXO` and `Input` concrete types in the `server/asset/{dcr,btc}` packages.

For sake of logging and messaging, I just moved the `FundingCoin` retrieval before the coin lock check as I those do not depend on each other (and I see no side effects of `FundingCoin that could be a problem for coins already used in a swap that would have been caught first by the coin locker).